src/control/lib/control/pool.go

```diff
@@ -194,6 +194,45 @@ func PoolCreate(ctx context.Context, rpcClient UnaryInvoker, req *PoolCreateReq)
 	}, nil
 }
 
+type (
+	// PoolResolveIDReq contains the parameters for a request to resolve
+	// a human-friendly pool identifier to a UUID.
+	PoolResolveIDReq struct {
+		msRequest
+		unaryRequest
+		HumanID string
+	}
+
+	// PoolResolveIDResp contains the result of a successful request.
+	PoolResolveIDResp struct {
+		UUID string
+	}
+)
+
+// PoolResolveID resolves a user-friendly Pool identifier into a UUID for use
+// in subsequent API requests.
+func PoolResolveID(ctx context.Context, rpcClient UnaryInvoker, req *PoolResolveIDReq) (*PoolResolveIDResp, error) {
```

Comment:
From a conceptual level, I'm wondering, do we really want to add an additional control API level RPC to resolve the label? I'd lean toward hiding the UUID resolution on the server side, though that raises more questions of how to update the input structures of existing API calls.

Reply:
Right. I didn't love this approach but I also didn't love adding ambiguity to the existing messages by overloading the uuid field or adding complexity by forcing every pool request message to have uuid or humanid.

In practice, I don't know how often non-UUID identifiers will actually be used, so I figured that I would go with the less-invasive approach for now. We can always optimize it later without any externally-visible changes.

Reply:
Hmm, that's fair. I'm fine with leaving it as-is for now.